libs/daemon/daemon.go

```diff
@@ -0,0 +1,120 @@
+package daemon
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"strconv"
+)
+
+const DatabricksCliParentPid = "DATABRICKS_CLI_PARENT_PID"
+
+type Daemon struct {
+	// If provided, the child process's pid will be written in the file at this
+	// path.
+	PidFilePath string
+
+	// Environment variables to set in the child process.
+	Env []string
+
+	// Path to executable to run. If empty, the current executable is used.
+	Executable string
+
+	// Arguments to pass to the child process.
+	Args []string
+
+	// Log file to write the child process's output to.
+	LogFile string
+
+	logFile *os.File
+	cmd     *exec.Cmd
+	stdin   io.WriteCloser
+}
+
+func (d *Daemon) Start() error {
+	cli, err := os.Executable()
+	if err != nil {
+		return err
+	}
```

Comment:
This code defaults `d.Executable` so it can be conditional on `d.Executable` being empty.

Reply:
Fixed in https://github.com/databricks/cli/pull/2376